File: 240623/문자열 놀이/play-with-string.py
s_str, str_q = input().split()
questions = int(str_q)
ques = [input().split() for _ in range(questions)]

s_str_lst = list(s_str)
for n, a, b in ques:
    if n == '1':
        a_int, b_int = int(a), int(b)
        s_str_lst[a_int - 1], s_str_lst[b_int - 1] = s_str_lst[b_int - 1], s_str_lst[a_int - 1]
    elif n == '2':
        for idx, s_str in enumerate(s_str_lst):
            if s_str == a:
                s_str_lst[idx] = b

    print("".join(s_str_lst))

File: 251206/문자열 놀이/play-with-string.cpp
#include <iostream>
#include <string>
using namespace std;

int main() 
{
    string s_str;
    int questions;
    cin >> s_str >> questions;

    for (int qi = 0; qi < questions; qi++) 
    {
        string n, a, b;
        cin >> n >> a >> b;

        if (n == "1") 
        {
            int a_int = stoi(a);
            int b_int = stoi(b);
            swap(s_str[a_int - 1], s_str[b_int - 1]);
        } 
        else if (n == "2") 
        {
            char from = a[0];
            char to = b[0];
            for (int i = 0; i < (int)s_str.size(); i++) 
            {
                if (s_str[i] == from) {
                    s_str[i] = to;
                }
            }
        }

        cout << s_str << '\n';
    }

    return 0;
}
